templates/CMakeLists.txt.template

```diff
@@ -150,6 +150,20 @@
       private_libs.append("-l" + lib_name)
     return private_libs
 
+  def lib_type_for_lib(lib_name):
+    """Returns STATIC/SHARED to force a static or shared lib build depending on the library."""
+    # grpc_csharp_ext is loaded by C# runtime and it
+    # only makes sense as a shared lib.
+    if lib_name in ['grpc_csharp_ext']:
+      return ' SHARED'
+
+    # upb always compiles as a static library on Windows
+    elif lib_name in ['upb','upb_collections_lib','upb_json_lib','upb_textformat_lib']:
```

Comment:
note that in the future we'll need a better way of telling which are the "upb" libraries than just listing them in multiple places in the CmakeLists.txt.tempate.
FWIW, there should be single place that defines sth like `is_upb_library` and use that from elsewhere. Unfortunately, since our upb cmake dependency is in flux, we're not there yet.

Reply:
I can add that function. Would the following predicate suffice?
```text
lib_name == 'upb' or lib_name.startswith('upb_')
```